src/components/Modals/Receive/ReceiveRouter.tsx

```diff
@@ -15,40 +15,45 @@ type ReceiveRouterProps = {
   assetId?: AssetId
   accountId?: AccountId
 }
-export const ReceiveRouter = ({ assetId, accountId }: ReceiveRouterProps) => {
+export const ReceiveRouter: React.FC<ReceiveRouterProps> = ({ assetId, accountId }) => {
   const asset = useAppSelector(state => selectAssetById(state, assetId ?? ''))
   const [selectedAsset, setSelectedAsset] = useState<Asset | undefined>(asset)
-  const location = useLocation()
-  const history = useHistory()
+  const navigate = useNavigate()
 
   const handleAssetSelect = useCallback(
     (assetId: AssetId) => {
       const _asset = selectAssetById(store.getState(), assetId)
       setSelectedAsset(_asset)
-      history.push(ReceiveRoutes.Info)
+      navigate(ReceiveRoutes.Info)
     },
-    [history],
+    [navigate],
   )
 
   useEffect(() => {
     if (!selectedAsset && !asset) {
-      history.push(ReceiveRoutes.Select)
+      navigate(ReceiveRoutes.Select)
     } else if (asset) {
       setSelectedAsset(asset)
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
+  const receiveInfoElement = useMemo(
+    () => (selectedAsset ? <ReceiveInfo asset={selectedAsset} accountId={accountId} /> : null),
+    [selectedAsset, accountId],
+  )
+
+  const selectAssetRouterElement = useMemo(
+    () => <SelectAssetRouter onClick={handleAssetSelect} />,
+    [handleAssetSelect],
+  )
+
   return (
     <AnimatePresence mode='wait' initial={false}>
-      <Switch location={location} key={location.key}>
-        <Route path={ReceiveRoutes.Info}>
-          {selectedAsset ? <ReceiveInfo asset={selectedAsset} accountId={accountId} /> : null}
-        </Route>
-        <Route path={ReceiveRoutes.Select}>
-          <SelectAssetRouter onClick={handleAssetSelect} />
-        </Route>
-      </Switch>
+      <Routes>
```

Comment:
Ditto, same as above, this works - gut feel is the `<Dialog />` wrapper escape hatches react-router global context or something, which makes those work. Weird, but hey we'll take it!

Reply:
Another one bites the dust in https://github.com/shapeshift/web/pull/9275 - but unless necessary, prefer not touching things that do as this PR is a monster already and in a working state 🙏🏽 

Agree would be nice to follow-up with all the remaining offenders however (not like there are many), will be easy to regression test in isolation then.